### PR TITLE
feat--add: array manipulation library exercise

### PR DESCRIPTION
**Pull request by jlabausa**
added array Manipulation library.

### CODE REVIEW

File: examples/02-javascript-basics/array_manipulation_library.js [hunk 1]
```diff
@@ -11,12 +11,62 @@ Join - join the array elements into a string given a parameter
...
+	reduce: function (values, test, initial_value) {
+		var result = [];
+		var total = 0;
+
+		for(var i = 0; i < values.length; i++) {
+			total = test(values[i], total);
+		}
+		
+		// Checks if there is initial value
+		if(initial_value || initial_value == 0) {
```
Comment (adimasuhid) [inline examples/02-javascript-basics/array_manipulation_library.js]: Prefer `===` when applicable

File: examples/02-javascript-basics/array_manipulation_library.js [hunk 2]
```diff
@@ -11,12 +11,62 @@ Join - join the array elements into a string given a parameter
...
+			total = total + initial_value
+		} else {
+			if(values.length == 0) {
```
Comment (adimasuhid) [inline examples/02-javascript-basics/array_manipulation_library.js]: This seems to be an `else if {}`. Prefer the syntax `else if {}`  when applicable.

File: examples/02-javascript-basics/array_manipulation_library.js [hunk 3]
```diff
@@ -11,12 +11,62 @@ Join - join the array elements into a string given a parameter
...
+				return result; // returns empty array if no init value and total
+			}
+		}
+
+		return total;
+	},
+	head: function (values) {
+		return values[0];
```
Comment (adimasuhid) [inline examples/02-javascript-basics/array_manipulation_library.js:51]: 👍

File: examples/02-javascript-basics/array_manipulation_library.js [hunk 4]
```diff
@@ -11,12 +11,62 @@ Join - join the array elements into a string given a parameter
...
+	},
+	tail: function (values) {
+		return values[values.length - 1];
+	},
+	join: function (values, addon) {
```
Comment (adimasuhid) [inline examples/02-javascript-basics/array_manipulation_library.js:56]: 👍

File: examples/02-javascript-basics/array_manipulation_library.js (hunk 2 above)
Comment (jlabausa) [reply examples/02-javascript-basics/array_manipulation_library.js]: Noted and applied :)

File: examples/02-javascript-basics/array_manipulation_library.js (hunk 1 above)
Comment (jlabausa) [reply examples/02-javascript-basics/array_manipulation_library.js]: Noted and applied :)

**Comment by adimasuhid** [comment]
Closing as this is already good

**Pull request closed without merging**